components/brave_viewer/browser/core/brave_viewer_component_installer.cc

```diff
@@ -0,0 +1,144 @@
+#include "components/component_updater/component_installer.h"
+#include "components/component_updater/component_updater_service.h"
+#include "crypto/sha2.h"
+
+namespace brave_viewer {
+
+namespace {
+
+constexpr size_t kHashSize = 32;
+const char kBraveViewerComponentName[] = "Brave Viewer Files";
+const char kBraveViewerComponentId[] = "TODO";
+const char kBraveViewerComponentBase64PublicKey[] = "TODO";
+
+}  // namespace
+
+class BraveViewerComponentInstallerPolicy
+    : public component_updater::ComponentInstallerPolicy {
+ public:
+  BraveViewerComponentInstallerPolicy();
+
+  BraveViewerComponentInstallerPolicy(
+      const BraveViewerComponentInstallerPolicy&) = delete;
+  BraveViewerComponentInstallerPolicy& operator=(
+      const BraveViewerComponentInstallerPolicy&) = delete;
+
+  // component_updater::ComponentInstallerPolicy
+  bool SupportsGroupPolicyEnabledComponentUpdates() const override;
+  bool RequiresNetworkEncryption() const override;
+  update_client::CrxInstaller::Result OnCustomInstall(
+      const base::Value::Dict& manifest,
+      const base::FilePath& install_dir) override;
+  void OnCustomUninstall() override;
+  bool VerifyInstallation(const base::Value::Dict& manifest,
+                          const base::FilePath& install_dir) const override;
+  void ComponentReady(const base::Version& version,
+                      const base::FilePath& path,
+                      base::Value::Dict manifest) override;
+  base::FilePath GetRelativeInstallDir() const override;
+  void GetHash(std::vector<uint8_t>* hash) const override;
+  std::string GetName() const override;
+  update_client::InstallerAttributes GetInstallerAttributes() const override;
+
+ private:
+  const std::string component_id_;
+  const std::string component_name_;
+  uint8_t component_hash_[kHashSize];
+};
+
+BraveViewerComponentInstallerPolicy::BraveViewerComponentInstallerPolicy()
+    : component_id_(kBraveViewerComponentId),
+      component_name_(kBraveViewerComponentName) {
+  // Generate hash from public key.
+  std::string decoded_public_key;
+  base::Base64Decode(kBraveViewerComponentBase64PublicKey, &decoded_public_key);
+  crypto::SHA256HashString(decoded_public_key, component_hash_, kHashSize);
+}
+
+bool BraveViewerComponentInstallerPolicy::
+    SupportsGroupPolicyEnabledComponentUpdates() const {
+  return true;
+}
+
+bool BraveViewerComponentInstallerPolicy::RequiresNetworkEncryption() const {
+  return false;
+}
+
+update_client::CrxInstaller::Result
+BraveViewerComponentInstallerPolicy::OnCustomInstall(
+    const base::Value::Dict& manifest,
+    const base::FilePath& install_dir) {
+  return update_client::CrxInstaller::Result(0);
+}
+
+void BraveViewerComponentInstallerPolicy::OnCustomUninstall() {}
+
+void BraveViewerComponentInstallerPolicy::ComponentReady(
+    const base::Version& version,
+    const base::FilePath& path,
+    base::Value::Dict manifest) {
+  BraveViewerService::GetInstance()->LoadNewComponentVersion(path);
```

Comment:
I wonder if this will be a problem for iOS until we have flags working. cc @Brandon-T